services/datalad/datalad_service/common/annex.py

```diff
@@ -196,7 +196,7 @@ def get_repo_files(dataset, branch='HEAD'):
             filename = symlinkFilenames[(index - 1) // 2]
             file_id = compute_file_hash(key, filename)
             files.append({'filename': filename, 'size': int(
-                size), 'id': file_id, 'key': key, 'urls': []})
+                size), 'id': file_id, 'key': key, 'urls': [], 'annexed': True})
```

Comment:
Tried seeing if we could do this without passing the key value back to the client because it increases the size of the files response but it is required to keep the Apollo cache id unique for the same content in different trees. We might want to add some additional pagination support to improve the performance for larger datasets but I think this works for now.